streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -728,8 +733,11 @@ private void subscribeConsumer() {
         }
     }
 
-    public void resizeCache(final long size) {
-        cacheResizeSize.set(size);
+    public void resizeCacheOrBufferMemory(final long size, final boolean cacheResize) {
```

Comment:
See the other comment: if we always size the two at the same time, we can just make one function call instead of two calls with a second boolean flag.